drift_dev/lib/src/analysis/resolver/file_analysis.dart

```diff
@@ -71,6 +71,18 @@ class FileAnalyzer {
                     (e) => e is DefinedSqlQuery || e is DriftSchemaElement);
               })
               .whereType<DriftElement>()
+              .where((e){
+                // Exclude any private tables (name prefixed with "_") that do not
+                // reside in the same library as the DriftDatabase.
+                // Failure to exclude these, can generate dart code which references
+                // classes that cannot be legally accessed - and will not compile.
+                // Private classes in residing in the same library are allowed, as
+                // per dart language accessibility rules.                
+                if ((e is DriftTable) && e.baseDartName.startsWith('_')) {
```

Comment:
> I couldn't use the 'id' property to detect "private scope" as suggested, as its the name of the dart class, not the table name, thats the issue

How is this an issue again? The id should point to the definition of the class, and we want to exclude it if it's private, right?

The `e is DriftTable` condition is a bit unfortunate here because it means that this won't apply to views. So I'm wondering if we can use the id or `e.entityInfoName` as an alternative.

Reply:
Hi, apologies, I may not have explained what I found very well the first time.

I couldn't use 'id.name' to detect "privateness", because id.name appeared to be the DataClassName, not the name of the table class.

For example, if I change my fix, to compare 'e.id.name' instead of 'e.baseDartName', then the following will fail to be picked up as a private table class (notice how @DataClassName does not have the underscore, this is what goes into id.name).
I did consider - why would someone have a private table, but a public dataclass? But I really don't know, but if they do - it will just break, which I didn't like.

```
@DataClassName('NotPartOfDatabase')
class _NotPartOfDatabase extends Table {
  IntColumn get id => integer()();
}
```

Your suggestion of `entityInfoName` looks viable, that will cover both tables and views, and I just need to look for '$_' instead of '_'.

The predicate then becomes:
```
                if (e is DriftElementWithResultSet && e.entityInfoName.startsWith('\$_')) {
                  return e.id.libraryUri == element.id.libraryUri;
                }
```

Do you think that's the way to go?
I've committed it with some enhancements to the integration test for your consideration.

Thanks again Simon.

Reply:
Just an aside @simolus3 - thanks for your patience. I'm doing my best to understand the drift_dev internals with the time I have available to me, to produce something fitting of the level of quality you've achieved in drift. I appreciate that this is probably pulling your away from your day-job too.